sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2792,6 +2792,51 @@ object GpuOverrides {
           aggBuffer.copy(dataType = c.dataType)(aggBuffer.exprId, aggBuffer.qualifier)
         }
       }),
+    expr[ApproximatePercentile](
+      "Approximate percentile",
+      ExprChecks.fullAggAndProject(
+        // note that output can be single number or array depending on whether percentiles param
+        // is a single number or an array
+        TypeSig.numeric + TypeSig.DECIMAL_64 +
+          TypeSig.ARRAY.nested(TypeSig.numeric + TypeSig.DECIMAL_64),
+        TypeSig.ARRAY.nested(TypeSig.all),
```

Comment:
This is not correct.

The `inputTypes` for ApproximatePercentile are

```
  override def inputTypes: Seq[AbstractDataType] = {
    // Support NumericType, DateType and TimestampType since their internal types are all numeric,
    // and can be easily cast to double for processing.
    Seq(TypeCollection(NumericType, DateType, TimestampType),
      TypeCollection(DoubleType, ArrayType(DoubleType, containsNull = false)), IntegralType)
  }
```

The output might or might not be in an array, depending on the percentages passed in.

```
ExprChecks.groupByAggOnly(
    TypeSig.gpuNumeric + TypeSig.ARRAY.nested(TypeSig.gpuNumeric),
    TypeSig.numeric + TypeSig.DATE + TypeSig.TIMESTAMP + TypeSig.ARRAY.nested(TypeSig.numeric + TypeSig.DATE + TypeSig.TIMESTAMP),
    Seq(
      ParamCheck("input", TypeSig.gpuNumeric, TypeSig.numeric + TypeSig.DATE + TypeSig.TIMESTAMP),
      ParamCheck("percentage", TypeSig.DOUBLE + TypeSig.ARRAY.nested(TypeSig.DOUBLE), TypeSig.DOUBLE + TypeSig.ARRAY.nested(TypeSig.DOUBLE)),
      ParamCheck("accuracy", TypeSig.INT, TypeSig.INT))),
```

Reply:
I've updated the signatures but I don't see a `groupByAggOnly` method on ExprChecks